source/nodejs/adaptivecards-ui-testapp/package.json

```diff
@@ -36,6 +36,18 @@
 		"adaptive-expressions": "^4.11.0",
 		"adaptivecards": "^3.0.0-beta.1",
 		"adaptivecards-controls": "^0.9.0",
-		"adaptivecards-templating": "^2.3.0-alpha.0"
+		"adaptivecards-templating": "^2.3.0-alpha.0",
+		"copy-webpack-plugin": "^9.0.1",
+		"dotenv-webpack": "^7.0.3",
+		"html-webpack-plugin": "^5.4.0",
+		"mini-css-extract-plugin": "^2.4.3",
+		"typescript": "^4.4.4",
+		"webpack": "^5.60.0",
+		"webpack-cli": "^4.9.1",
+		"webpack-concat-files-plugin": "^0.5.2",
+		"webpack-dev-server": "^4.3.1",
+		"ts-loader": "^9.2.6",
+		"css-loader": "^6.4.0",
+		"style-loader": "^3.3.1"
```

Comment:
these dependencies should already be available in `source/nodejs`... rather than duplicating the dependencies here, you should just `npm i` in the parent directory before building.

removal of these dependencies was intentional to prevent us from managing version updates across a bunch of diverging packages...

Reply:
To make the test app consumes the built package, this project is copied to source/ and run outside of lerna. so I need to dup them

Reply:
you shouldn't have to copy this outside of `source/nodejs`... you should be able to do something like:

1. Build and pack the target packages
1. Save packed `tgz` files outside of source (a temp directory should suffice)
2. `git clean -dxf`
3. `npm i` in `source/nodejs`
4. `npm i /path/to/saved/packs/*.tgz && npm i && npm run build` in `source/nodejs/adaptivecards-ui-testapp`

Managing each of our individual projects' npm dependencies is a pain, so I'd really like to avoid duplicating them to support this one CI task.

Reply:
It will not catch the bug like missing `swiper`. It only happens when command is run outside of lerna workspace

Reply:
`lerna` only comes into play when you `npx lerna bootstrap`. when you `git clean -dxf`, it will delete any cross-linking that `lerna` does:

![image](https://user-images.githubusercontent.com/16614499/148862804-aff8a4ec-6ca2-4660-8e87-34afc17ae70f.png)

and then after doing an `npm i swiper@latest` in `source/nodejs/adaptivecards-ui-testapp`:

![image](https://user-images.githubusercontent.com/16614499/148862989-0ded8dcd-a139-4c5e-be00-94c1f486343c.png)